library/src/compute/construct.rs

```diff
@@ -179,6 +181,174 @@ cast_from_value! {
     },
 }
 
+/// Create a new datetime.
+///
+/// You can specify the [datetime]($type/datetime) using a year, month, day,
+/// hour, minute, and second.
+///
+/// ## Example
+/// ```example
+/// #datetime(
+///   year: 2012,
+///   month: 8,
+///   day: 3,
+/// ).display()
```

Comment:
I'm still thinking whether the `.display()` should happen automatically when interpolating into markup. (Maybe also for counters and state.) Currently it shows `repr` because `Value::display` falls back to that when no suitable display is available. But here, there is one.

Reply:
We could do that, but as far as I understand this would always just show the default date/time format, there would be no way to change this with a set command, right? So not sure how useful that would be. After all, you can just do something like `#let my-date = datetime.today().display("my-format")` and then just reuse the variable.

Reply:
Yes, it would show the default format for now. (Maybe we can improve that in the future though.) I think it's a better to show the default format than the repr in this case (you can of course still customize it with `display`. You can always get the repr explicitly with `repr`.

Reply:
Yeah, maybe it's best to leave that for another PR.